core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -518,32 +522,50 @@ public long getUsedBytes() {
 
     List<WorkerInfo> workerInfoList = new ArrayList<>();
     for (MasterWorkerInfo worker : selectedLiveWorkers) {
-      synchronized (worker) {
-        workerInfoList.add(worker.generateWorkerInfo(options.getFieldRange(), true));
-      }
+      // extractWorkerInfo handles the locking internally
+      workerInfoList.add(extractWorkerInfo(worker, options.getFieldRange(), true));
     }
     for (MasterWorkerInfo worker : selectedLostWorkers) {
-      synchronized (worker) {
-        workerInfoList.add(worker.generateWorkerInfo(options.getFieldRange(), false));
-      }
+      // extractWorkerInfo handles the locking internally
+      workerInfoList.add(extractWorkerInfo(worker, options.getFieldRange(), false));
     }
     return workerInfoList;
   }
 
+  /**
+   * Lock the {@link MasterWorkerInfo} properly and convert it to a {@link WorkerInfo}
+   * */
+  private WorkerInfo extractWorkerInfo(MasterWorkerInfo worker,
+                                       Set<GetWorkerReportOptions.WorkerInfoField> fieldRange,
+                                       boolean isLiveWorker) {
+    worker.getMetaLock().readLock().lock();
+    worker.getUsageLock().readLock().lock();
+    try {
+      return worker.generateWorkerInfo(fieldRange, isLiveWorker);
+    } finally {
+      worker.getUsageLock().readLock().unlock();
+      worker.getMetaLock().readLock().unlock();
+    }
+  }
+
   @Override
   public List<WorkerLostStorageInfo> getWorkerLostStorage() {
     List<WorkerLostStorageInfo> workerLostStorageList = new ArrayList<>();
     for (MasterWorkerInfo worker : mWorkers) {
-      synchronized (worker) {
+      worker.getUsageLock().readLock().lock();
+      try {
         if (worker.hasLostStorage()) {
           Map<String, StorageList> lostStorage = worker.getLostStorage().entrySet()
-              .stream().collect(Collectors.toMap(Map.Entry::getKey,
-                  e -> StorageList.newBuilder().addAllStorage(e.getValue()).build()));
+                  .stream().collect(Collectors.toMap(Map.Entry::getKey,
+                          e -> StorageList.newBuilder().addAllStorage(e.getValue()).build()));
           workerLostStorageList.add(WorkerLostStorageInfo.newBuilder()
-              .setAddress(GrpcUtils.toProto(worker.getWorkerAddress()))
-              .putAllLostStorage(lostStorage).build());
+                  .setAddress(GrpcUtils.toProto(worker.getWorkerAddress()))
+                  .putAllLostStorage(lostStorage).build());
```

Comment:
revert unnecessary indent change

Reply:
Will do at the end of the review. Once we are settled on the implementations I will make sure the checkstyle and the style are happy.